app/lib/identity_tijuana/member_sync.rb

```diff
@@ -6,25 +6,55 @@ def self.export_member(member, sync_id)
       begin
         return if member.ghosting_started?
 
-        ext_id = MemberExternalId.find_by(system: 'tijuana', member: member)
-        user_id = ext_id.external_id if ext_id.present?
-        if user_id.present?
-          user = User.find_by(id: user_id)
-          ext_id.destroy! if user.blank?
+        ext_ids = MemberExternalId.where(system: 'tijuana', member: member).to_a
+        # Destroy any dangling external ID references, remove any
+        # external ids from the list if not found
+        ext_ids.select do |ext_id|
+          user = User.find_by(id: ext_id.external_id.to_i)
```

Comment:
Maybe premature as a question, perhaps treat the following as comment.

I haven't seen how soft delete is handled, so not sure how will this affect TJ users with `deleted_at`.
If they are not returned here, and we simply unlink them from ID member, we may end up with invalid data.

On the other hand, do you think it's sufficient just to log unlinking the records via external ID, or should we consider adding it to `active record audit`?

Reply:
> I haven't seen how soft delete is handled, so not sure how will this affect TJ users with deleted_at.
If they are not returned here, and we simply unlink them from ID member, we may end up with invalid data.

TJ uses [paranoia](https://github.com/rubysherpas/paranoia) to implement this. Since neither Id or this engine imports that package, and `app/models/identity_tijuana/user.rb` doesn't override the default scope to manually implement it, the `User.find_by` call will return soft-deleted users.

> On the other hand, do you think it's sufficient just to log unlinking the records via external ID, or should we consider adding it to active record audit?

Yeah audit would be ideal - did you look into that after the convo this morning? I do want all operations logged here as well though - there's a huge benefit in having everything in the one place (coralogix) for being able to debug issues.

Reply:
> Yeah audit would be ideal - did you look into that after the convo this morning?

yeas, ActiveRecord audit doesn't log any changes relating to `MemberExternalId`; only  changes to the following are logged:
* PhoneNumber
* Member
* Address
* MemberSubscription
* CustomField

Reply:
Yeah maybe we need to add that then.